Polyfills/XMLHttpRequest/Source/XMLHttpRequest.cpp

```diff
@@ -180,7 +181,14 @@ namespace Babylon::Polyfills::Internal
     {
         try
         {
-            m_request.Open(MethodType::StringToEnum(info[0].As<Napi::String>().Utf8Value()), info[1].As<Napi::String>().Utf8Value());
+            auto inputURL{info[1].As<Napi::String>()};
+            // If the input URL contains any true % characters, encode them as %25
+            auto encodedPercentURL{Napi::String::New(info.Env(), EncodePercent(inputURL.Utf8Value()))};
```

Comment:
I realized that the only way to safely encode a URL is to start from a fully decoded one. Otherwise, you run the risk of re-encoding "%20" as "%2520.". To implement this, I came up with this solution:
1. If there's a standalone % character in the URL, encode it as %25.
2. Use decodeURI to get a truly decoded URL. This URL will have *nothing* percent encoded. If a % exists, it's because there's a real % character.
3. Finally, use encodeURI to get an encoded URL.

This method still primarily uses decodeURI/encodeURI, which are standards. The only custom code is for the first step, to turn % into %25 when it's not followed by hex characters. I think since this function is so simple, the risk of bugs is fairly low.

I added several tests which represent pretty much the worst possible URLs that I could imagine, and this solution handles them correctly, so it seems pretty safe.